src/argilla_sdk/records/_dataset_records.py

```diff
@@ -338,16 +334,29 @@ def from_json(self, path: Union[Path, str]) -> "DatasetRecords":
             DatasetRecords: The DatasetRecords object created from the disk path.
 
         """
-        with open(path, "r") as f:
-            records = json.load(f)
-        self.update(records=[_dict_to_record(record) for record in records])
+        records = self()._records_from_json(path=path)
+        self.update(records)
```

Comment:
@frascuchon What are you thoughts on returning List[Record] here? I struggled with inheritance and making it so the mixins only dealt with io.

Reply:
I think we can review the mixin definition and simplify/unify a bit if we focus on converting records into something and something into records.

Let me write a draft. I will share it with you and we can discuss it.